azext_iot/tests/conftest.py

```diff
@@ -335,6 +337,8 @@ def get_context_path(base_path, *paths):
 @pytest.fixture
 def mocked_response():
     with responses.RequestsMock() as rsps:
+        on_request_with_no_retry = partial(rsps._on_request, retries=Retry(0, read=False))
+        rsps._on_request = on_request_with_no_retry
```

Comment:
Any better way to do this without having to access private attributes?

Reply:
I will ask: https://github.com/getsentry/responses/issues/621